experimental/PropertyDDS/packages/property-binder/src/data_binder/internal_utils.js

```diff
@@ -551,7 +551,8 @@ const recursivelyVisitHierarchy = function(
             if (in_dataBindingTreeNode) {
               in_dataBindingTreeNode = in_dataBindingTreeNode.getNodeForTokenizedPath([id]);
             }
-            const subpath = (in_path === '/') ? '/' + id : in_path + '.' + id;
+            const typeidSplit = TypeIdHelper.extractContext(in_propertyElement.getProperty().getFullTypeid());
+            const subpath = (in_path === '/') ? '/' + id : typeidSplit.context !== 'single' ? in_path + `[${id}]` : in_path + '.' + id;
```

Comment:
Suggestion for future PRs: You may find ternary operators to be more readable when formatted as below.
```ts
const subpath = (in_path === '/')
    ? '/' + id
    : typeidSplit.context !== 'single'
        ? in_path + `[${id}]`
        : in_path + '.' + id;
```

Reply:
I think id should probably be escaped here.